JunYoung/week3/src/components/BottomSheetModal.tsx

```diff
@@ -0,0 +1,126 @@
+import React, { useEffect } from "react";
+import {
+  View,
+  Text,
+  StyleSheet,
+  TouchableOpacity,
+  Alert,
+  Platform,
+} from "react-native";
+import Modal from "react-native-modal"; // 모달 컴포넌트
+import * as ImagePicker from "expo-image-picker"; // 이미지 피커 라이브러리
+import { CamBtn, GallBtn } from "../assets/icons"; // 아이콘 컴포넌트
+import theme from "../theme"; // 테마 설정
+import { BottomSheetModalProps, ImagePickerResult } from "../types"; // 타입 정의
+
+// BottomSheetModal 컴포넌트 정의
+const BottomSheetModal: React.FC<BottomSheetModalProps> = ({
+  isVisible,
+  onClose,
+  onImageSelect,
+}) => {
+  // 컴포넌트가 마운트될 때 권한 요청
+  useEffect(() => {
+    (async () => {
+      if (Platform.OS !== "web") {
+        const { status } =
+          await ImagePicker.requestMediaLibraryPermissionsAsync();
+        if (status !== "granted") {
+          Alert.alert(
+            "죄송합니다. 이 기능을 사용하려면 카메라 롤 권한이 필요합니다!"
+          );
+        }
+      }
+    })();
+  }, []);
+
+  const handleCameraPress = async () => {
+    const { status } = await ImagePicker.requestCameraPermissionsAsync();
+    if (status !== "granted") {
+      Alert.alert("죄송합니다. 이 기능을 사용하려면 카메라 권한이 필요합니다!");
+      return;
+    }
+
+    const result = await ImagePicker.launchCameraAsync({
+      mediaTypes: ImagePicker.MediaTypeOptions.Images,
+      quality: 1, // 이미지 품질 설정 (0 ~ 1)
+      allowsEditing: true, // 이미지 편집 허용 여부
+      aspect: [4, 3], // 이미지 비율 설정
+      base64: false, // base64 인코딩된 이미지 데이터 포함 여부
+    });
+
+    if (!result.canceled) {
+      console.log("Camera Result:", result);
+      onImageSelect([result.assets[0] as ImagePickerResult]); // 결과를 ImagePickerResult 타입으로 변환하여 전달
+    } else {
+      onImageSelect([]);
```

Comment:
```suggestion
      onImageSelect(result.canceled ? [result.assets[0] as ImagePickerResult] : []);
```
`handleCameraPress` 호출 시 항상 onImageSelect를 호출하는거라면 위 코드처럼 한줄로 작성할 수도 있겠네요!

Reply:
앗 급박해서 코드를 간략하게 하는데 신경을 못썼습니다 ㅜㅜ

지적해주셔서 감사해요 :)